src/pygeosx/unitTests/dataRepositoryTests/test_pygeosxDataRepository.py

```diff
@@ -0,0 +1,181 @@
+                              (node_manager + 'ghostRank'),
+                              (cell_blocks + 'elementVolume')])
+    def test_wrapper_write(self, target_key, geosx_problem):
+        """
+        Test writing to the wrappers
+
+        Args:
+            target_key (str): Key value for the target wrapper
+            geosx_proplem (pygeosx.problem): The problem handle
+        """
+        # yapf: enable
+        # Access and record the initial value
+        x = geosx_problem.get_wrapper(target_key).value()
+        if hasattr(x, "set_access_level"):
+            x.set_access_level(pylvarray.MODIFIABLE, pylvarray.CPU)
+        if hasattr(x, "to_numpy"):
+            x = x.to_numpy()
+        x_init = x.copy()
+
+        # Modify the value
+        x += 1
+        del x
+
+        # Get another copy of the wrapper
+        x = geosx_problem.get_wrapper(target_key).value()
+        if hasattr(x, "to_numpy"):
+            x = x.to_numpy()
+
+        # Check to see if the value changed appropriately
+        dx = x_init + 1 - x
+        if isinstance(dx, np.ndarray):
+            dx = np.sum(dx)
+        dx = float(dx)
+
+        assert dx < 1e-10
+
+    # yapf: disable
+    @pytest.mark.parametrize('target_key',
+                             [(node_manager + 'ReferencePosition'),
+                              (node_manager + 'ghostRank'),
+                              (cell_blocks + 'elementVolume')])
+    def test_wrapper_read_only_error(self, target_key, geosx_problem):
+        """
+        Test writing to read-only wrappers.
+        Note: this error does not get called on certain scalar wrappers (time, cycle)
+
+        Args:
+            target_key (str): Key value for the target wrapper
+            geosx_proplem (pygeosx.problem): The problem handle
+        """
+        # yapf: enable
+        x = geosx_problem.get_wrapper(target_key).value()
+        if hasattr(x, "to_numpy"):
+            print('Test_a')
+            x = x.to_numpy()
+
+        with pytest.raises(ValueError):
+            x += 1
+
+    def test_problem_run(self, geosx_problem):
+        """
+        Test advancing the problem one step
+        """
+        status = pygeosx.run()
+        assert status == pygeosx.READY_TO_RUN
+        group = geosx_problem.get_group('Events', None)
+        ta = group.get_wrapper('time').value()[0].copy()
+        ca = group.get_wrapper('cycle').value()[0].copy()
+
+        status = pygeosx.run()
+        assert status == pygeosx.READY_TO_RUN
+        tb = group.get_wrapper('time').value()[0]
+        cb = group.get_wrapper('cycle').value()[0]
+        assert (ta + 1) == tb
+        assert (ca + 1) == cb
+
+    def test_problem_complete(self, geosx_problem):
+        """
+        Test running the problem to completion
+        """
+        max_iter = 100
+        ii = 0
+        while (pygeosx.run() != pygeosx.COMPLETED) and (ii <= max_iter):
+            ii += 1
+
+        assert ii != max_iter
```

Comment:
would it be possible to check that it actually takes exactly `10` iterations (deduced from the xml file, since there should be no other events happening)?